### DISCUSSION TITLE
GLib-GObject error thrown when loading plugin, bail out! server failed to start

### DISCUSSION THREAD

**Discussion by sukrosono** [hexojs/hexo#4974] [Question / Answer] [answered]
## Check List

Please check followings before submitting a new issue.

- [x] I have already read [Docs page](https://hexo.io/docs/) & [Troubleshooting page](https://hexo.io/docs/troubleshooting)
- [x] I have already searched existing issues and they are not help  to me
- [x] I examined error or warning messages and it's difficult to solve
- [x] Using [the latest](https://github.com/hexojs/hexo/releases) version of Hexo (run `hexo version` to check)
- [x] Node.js is higher than [minimum required version](https://hexo.io/docs/#Minimum-required-Node-js-version)

## Expected behavior
server running

## Actual behavior
```
18:03:56.260 DEBUG Writing database to /mnt/01CF064A1CD436D0/repository/enter-rebel.github.io/db.json
18:03:56.324 DEBUG Hexo version: 5.4.2
18:03:56.325 DEBUG Working directory: /mnt/01CF064A1CD436D0/repository/enter-rebel.github.io/
18:03:56.698 DEBUG Config loaded: /mnt/01CF064A1CD436D0/repository/enter-rebel.github.io/_config.yml
18:03:56.702 INFO  Validating config
18:03:59.294 DEBUG Plugin loaded: hexo-browsersync
18:03:59.295 DEBUG Plugin loaded: hexo-covers
18:03:59.295 DEBUG Plugin loaded: hexo-filter-github-emojis
18:03:59.295 DEBUG Plugin loaded: hexo-generator-category
18:03:59.295 DEBUG Plugin loaded: hexo-generator-archive
18:03:59.295 DEBUG Plugin loaded: hexo-generator-feed
18:03:59.296 DEBUG Plugin loaded: hexo-generator-search

(process:6033): GLib-GObject-WARNING **: 01:06:00.483: cannot register existing type 'gchar'
**
GLib-GObject:ERROR:../gobject/gvaluetypes.c:454:_g_value_types_init: assertion failed: (type == G_TYPE_CHAR)
Bail out! GLib-GObject:ERROR:../gobject/gvaluetypes.c:454:_g_value_types_init: assertion failed: (type == G_TYPE_CHAR)
Aborted (core dumped)
```
## How to reproduce?

* idk for sure.
* maybe it only for arch user, it happens after i update somethings.
* I am on manjaro with initramfs fall back mode

## Is the problem still there under "Safe mode"?
Yes, I am unable to start hexo server, but the hexo --version give result under safe mode flag.

## Environment & Settings

**Node.js & npm version(`node -v && npm -v`)**

```
node -v
v14.15.5
npm -v
8.7.0
```

**Your site `_config.yml`** (Optional)

```yaml
```

**Hexo and Plugin version(`npm ls --depth 0`)**

```
├── cson@7.20.0
├── hexo-browsersync@0.3.0
├── hexo-covers@2.0.0
├── hexo-filter-github-emojis@3.0.5
├── hexo-generator-archive@1.0.0
├── hexo-generator-category@1.0.0
├── hexo-generator-feed@3.0.0
├── hexo-generator-index@2.0.0
├── hexo-generator-search@2.4.3
├── hexo-generator-tag@1.0.0
├── hexo-images@1.0.1
├── hexo-renderer-ejs@1.0.0
├── hexo-renderer-marked@4.1.0
├── hexo-renderer-stylus@2.0.1
├── hexo-server@2.0.0
├── hexo-tag-wikipedia@1.0.1
├── hexo-wordcount@6.0.1
├── hexo@5.4.2
└── jquery@3.6.0
```

**Your package.json `package.json`**

```
{
  "hexo": {
    "version": "5.4.2"
  },
  "dependencies": {
    "hexo": "^5.4.2",
    "hexo-browsersync": "^0.3.0",
    "hexo-covers": "^2.0.0",
    "hexo-filter-github-emojis": "^3.0.4",
    "hexo-generator-archive": "^1.0.0",
    "hexo-generator-category": "^1.0.0",
    "hexo-generator-feed": "^3.0.0",
    "hexo-generator-index": "^2.0.0",
    "hexo-generator-search": "^2.4.2",
    "hexo-generator-tag": "^1.0.0",
    "hexo-renderer-ejs": "^1.0.0",
    "hexo-renderer-marked": "^4.0.0",
    "hexo-renderer-stylus": "^2.0.0",
    "hexo-server": "^2.0.0",
    "hexo-tag-wikipedia": "^1.0.0",
    "hexo-wordcount": "^6.0.1",
    "jquery": "^3.6.0"
  },
  "devDependencies": {
    "cson": "^7.20.0",
    "hexo-images": "^1.0.1"
  }
}
```
## hexo --version --safe
```
INFO  Validating config
hexo: 5.4.2
hexo-cli: 4.2.0
os: Linux 5.10.109-1-MANJARO linux x64
node: 14.15.5
v8: 8.4.371.19-node.18
uv: 1.40.0
zlib: 1.2.11
brotli: 1.0.9
ares: 1.16.1
modules: 83
nghttp2: 1.41.0
napi: 7
llhttp: 2.1.3
openssl: 1.1.1i
cldr: 37.0
icu: 67.1
tz: 2020a
unicode: 13.0
```

## Others
related to hexo last things I install are jquery and hexo-images.
The error is related to gastby I think, but idk where I use gatsby for sure.

**Comment by yoshinorin** [comment, marked as answer]
@sukrosono 
I assume the `hexo-images` depends on `sharp` and maybe it causes this issue. I don't know but `npm up && npm ls sharp` may effective.

> https://github.com/gatsbyjs/gatsby/issues/16957#issuecomment-534712756

**Reply by sukrosono** [reply]
```
ap.programming'site@0.0.0 /mnt/01CF064A1CD436D0/repository/enter-rebel.github.io
└─┬ hexo-images@1.0.1
  ├── sharp@0.26.3
  └─┬ smartcrop-sharp@2.0.5
    └── sharp@0.29.3
```
So I cant use hexo and hexo-images on my system, still looking a way out.
Basically, disabling the plugin will remove the error, I hope not to face this error on a real server. 

I like to notify the hexo-images developer, but I don't know who.

**Reply by yoshinorin** [reply]
@sukrosono 
> I like to notify the hexo-images developer, but I don't know who.

Not sure, but perhaps the repository is [here](https://github.com/sergeyzwezdin/hexo-images).
You can find who is the owner of the hexo-image if you see the `package.json` of it.

The `package.json` location is `<your_hexo_dir>/node_modules/hexo-images/package.json`.

**Reply by sukrosono** [reply]
@sergeyzwezgin @sergeyzwezdin does it work?

**Reply by sergeyzwezdin** [reply]
@sukrosono yes, it works for me.
Try `smartcrop-sharp` on the machine where it doesn't work.
